src/v/security/oidc_service.cc

```diff
@@ -41,6 +44,76 @@ template<typename... Args>
 
 using seastar::operator co_await;
 
+class probe {
+public:
+    using hist_t = log_hist_public;
+
```

Comment:
I didn't get the build failure locally, but the fix is something like:
```c++
    class auto_measure {
    public:
        explicit auto_measure(probe* p)
          : _p{p}
          , _m{p->_request_latency} {}
    public:
        void success() noexcept { _p->request_success(std::move(_m)); }
        void failed() noexcep`t { _p->request_error(std::move(_m)); }
    private:
        probe* _p;
        hist_t::measurement _m;
    };
```
And then used like:
```diff
-            probe->request_error(std::move(measure));
+            measure.failed();
```

Reply:
Change looks good, but I'm curious to know what the failure looked like - I don't see it anywhere.

Reply:
For example:
```c++
        if (response_body.failed()) {
            probe->report_error(std::move(measure));
            co_await return_exception(               // Note, this is really a return
              errc::metadata_invalid,
              "Failed to retrieve metadata: {}, error: {}",
              _discovery_url(),
              response_body.get_exception());
        }
        auto metadata = oidc::metadata::make(response_body.get());
        if (metadata.has_error()) {
            probe->report_error(std::move(measure));  // Use after move 
            co_await return_exception(
              metadata.assume_error(),
              "invalid response from discovery_url: {}, err: {}",
              _discovery_url(),
              metadata.assume_error().message());
        }
```

Reply:
Oh yeah, I understand the issue. I thought you meant it was flagged in CI somehow but not locally. Nbd.

Reply:
> I thought you meant it was flagged in CI somehow but not locally

That was the case: https://buildkite.com/redpanda/redpanda/builds/41244#018bd917-04ca-425d-85af-e02710a75b54/474-1023